scripts/tests/forest_cli_check.sh

```diff
@@ -1,45 +1,70 @@
 #!/usr/bin/env bash
 # This script is used to test the `forest-cli` commands that do not
 # require a running `forest` node.
+#
+# It's run in CI, and tests things that currently aren't tested (and don't make
+# sense to test) in our rust test harness.
+# This means things like fetching and validating snapshots from the web.
+#
 # It depends on the `forest-cli` binary being in the PATH.
 
-set -e
+set -euxo pipefail
 
 source "$(dirname "$0")/harness.sh"
 
-SNAPSHOT_DIRECTORY=$TMP_DIR/snapshots
-
-echo "Fetching params"
-$FOREST_CLI_PATH fetch-params --keys
-echo "Downloading zstd compressed snapshot without aria2"
-$FOREST_CLI_PATH --chain calibnet snapshot fetch --provider filecoin --compressed -s "$SNAPSHOT_DIRECTORY"
-echo "Downloading snapshot without aria2"
-$FOREST_CLI_PATH --chain calibnet snapshot fetch -s "$SNAPSHOT_DIRECTORY"
-echo "Cleaning up snapshots"
-$FOREST_CLI_PATH --chain calibnet snapshot clean -s "$SNAPSHOT_DIRECTORY" --force
-echo "Cleaning up snapshots again"
-$FOREST_CLI_PATH --chain calibnet snapshot clean -s "$SNAPSHOT_DIRECTORY" --force
-echo "Downloading zstd compressed snapshot"
-$FOREST_CLI_PATH --chain calibnet snapshot fetch --aria2 --provider filecoin --compressed -s "$SNAPSHOT_DIRECTORY"
-echo "Cleaning up database"
-$FOREST_CLI_PATH --chain calibnet db clean --force
-echo "Cleaning up database again"
-$FOREST_CLI_PATH --chain calibnet db clean --force
-
-echo "Downloading snapshot"
-$FOREST_CLI_PATH --chain calibnet snapshot fetch --aria2 -s "$SNAPSHOT_DIRECTORY"
-
-echo "Validating as mainnet snapshot"
-set +e
-$FOREST_CLI_PATH --chain mainnet snapshot validate "$SNAPSHOT_DIRECTORY"/*.car --force && \
-{
-    echo "mainnet snapshot validation with calibnet snapshot should fail";
-    exit 1;
+snapshot_dir=$(mktemp --directory)
+
+# return the first snapshot path listed by forest
+function pop-snapshot-path() {
+    forest-cli snapshot list --snapshot-dir "$snapshot_dir" \
+        | grep path \
+        | sed 's/^\tpath: //' \
+        | head --lines=1
+}
+
+function clean-snapshot-dir() {
+    rm --recursive --force --verbose -- "$snapshot_dir"
+    mkdir --parents -- "$snapshot_dir"
 }
-set -e
 
-echo "Validating as calibnet snapshot (uncompressed)"
-$FOREST_CLI_PATH --chain calibnet snapshot validate "$SNAPSHOT_DIRECTORY"/*.car --force
+# TODO(aatifsyed): I don't really understand what this does
```

Comment:
It just calls one of the subcommands that we know we can call and should not fail.